homeassistant/components/freebox/sensor.py

```diff
@@ -38,42 +38,57 @@ def state(self):
         """Return the state of the sensor."""
         return self._state
 
-    async def async_update(self):
-        """Fetch status from freebox."""
-        self._datas = await self._fbx.connection.get_status()
-
-
-class FbxRXSensor(FbxSensor):
-    """Update the Freebox RxSensor."""
-
-    _name = 'Freebox download speed'
-    _unit = 'KB/s'
-
     @property
     def unit_of_measurement(self):
         """Define the unit."""
         return self._unit
 
+    @property
+    def icon(self):
+        """Define the icon."""
+        return self._icon
+
     async def async_update(self):
         """Get the value from fetched datas."""
-        await super().async_update()
+        self._datas = self._hass.data[DATA_FREEBOX_SENSOR]
         if self._datas is not None:
             self._state = round(self._datas['rate_down'] / 1000, 2)
 
 
-class FbxTXSensor(FbxSensor):
+class FbxTXSensor(Entity):
     """Update the Freebox TxSensor."""
 
-    _name = 'Freebox upload speed'
-    _unit = 'KB/s'
+    def __init__(self, hass):
+        """Initialize the sensor."""
+        self._state = None
+        self._name = 'Freebox upload speed'
+        self._unit = 'KB/s'
```

Comment:
We can extract the string constants into module level constants which we return directly in the entity properties.